src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,322 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu/x64/jit_generator.hpp"
+#include <dnnl_types.h>
+#include "ie_common.h"
+#include <utility>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl::cpu;
+
+/**
+ * The RegistersPool is the base class for the IsaRegistersPool template:
+ *      template <x64::cpu_isa_t isa>
+ *      class IsaRegistersPool : public RegistersPool;
+ *
+ * The registers pool must be created by instantiating the IsaRegistersPool template, like the next:
+ *      RegistersPool::Ptr regPool = RegistersPool::create<isa>({
+ *          // the list of the registers to be excluded from pool
+ *          Reg64(Operand::RAX), Reg64(Operand::RCX), Reg64(Operand::RDX), Reg64(Operand::RBX),
+ *          Reg64(Operand::RSP), Reg64(Operand::RBP), Reg64(Operand::RSI), Reg64(Operand::RDI)
+ *      });
+ */
+class RegistersPool {
+public:
+    using Ptr = std::shared_ptr<RegistersPool>;
+    static constexpr int anyIdx = -1;
+
+    /**
+     * The scoped wrapper for the Xbyak registers.
+     * By creating it you are getting the register from the pool RegistersPool.
+     * It could be created by using constructor with RegistersPool as an argument, like the next:
+     *      const RegistersPool::Reg<Xbyak::Reg64> reg {regPool};
+     * The destructor will return the register to the pool. Or it could be returned manually:
+     *      reg.release();
+     * @tparam TReg Xbyak register class
+     */
+    template<typename TReg>
+    class Reg {
+        friend class RegistersPool;
+    public:
+        Reg() {
+            static_assert(std::is_same<TReg, Xbyak::Xmm>::value || std::is_same<TReg, Xbyak::Ymm>::value || std::is_same<TReg, Xbyak::Zmm>::value ||
+                          std::is_same<TReg, Xbyak::Reg8>::value || std::is_same<TReg, Xbyak::Reg16>::value ||
+                          std::is_same<TReg, Xbyak::Reg32>::value || std::is_same<TReg, Xbyak::Reg64>::value ||
+                          std::is_same<TReg, Xbyak::Opmask>::value,
```

Comment:
As for me, introducing a new type_traits when we could use standard, is not very good idea ...
Any new type_trait add an additional level of complexity

Reply:
Hm, that is more a matter o taste, so I would not argue here much. But,
```cpp
static_assert(is_any<TReg, Xbyak::Xmm, Xbyak::Ymm, Xbyak::Reg8>::value, "Error!!!");
```
looks just better. As for complexity, we can easily hide those type traits. But, it may be useful somewhere else as well. Anyway, I leave the decision here to you.

Reply:
Thank you for the hint. I found implementation for is_any_of and reused it in the static_assert.